### PR TITLE
Connect to api

### PR DESCRIPTION
**Pull request by atanasovRazvan**
Created server calls directly from the LandingPage component (no more mock data)

### CODE REVIEW

File: src/landing/LandingPage.js [hunk 1]
```diff
@@ -1,14 +1,91 @@
 import React, { useState } from 'react';
+import { gql, useLazyQuery, useMutation } from '@apollo/client';
 import {
   Paper, Tab, Tabs,
 } from '@material-ui/core';
 import LoginForm from './LoginForm';
 import './styles.css';
 import RegisterForm from './RegisterForm';
 
+const GET_USERS = gql`
```
Comment (ih90) [inline src/landing/LandingPage.js]: Not a change request but you may create a file to hold the GQL queries/mutation. Could be named "sources", for example.

File: src/landing/LandingPage.js [hunk 2]
```diff
@@ -1,14 +1,91 @@
...
+
+    if (queryError) {
+      console.log(queryError);
+      alert('Something went wrong');
+      return;
+    }
+
+    let loginStatus = false;
+    if (queryData) {
+      queryData.users.forEach((user) => {
```
Comment (ih90) [inline src/landing/LandingPage.js]: You could try to handle this with the help of Array.find()

File: src/landing/LandingPage.js [hunk 3]
```diff
@@ -1,14 +1,91 @@
...
+      });
+
+    if (mutationError) {
+      alert('Something went wrong...');
+      console.log(mutationError);
+    } else {
+      alert('Register successful! Redirecting to login...');
+    }
+  };
+
```
Comment (ih90) [inline src/landing/LandingPage.js]: Please check the useMutation hook API as we have better ways to handle errors or success (see onError, onComplete etc). https://www.apollographql.com/docs/react/data/mutations/

File: src/apollo.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import { ApolloClient, InMemoryCache } from '@apollo/client';
 
 const apolloClient = new ApolloClient({
-  uri: 'https://48p1r2roz4.sse.codesandbox.io', // TODO change here
+  uri: 'http://localhost:4001/gql', // TODO change here
```
Comment (ih90) [inline src/apollo.js]: Nitpick - try to remove the comments or update them as needed.

Comment (atanasovRazvan) [reply src/apollo.js]: removed comments

File: src/landing/LandingPage.js (hunk 1 above)
Comment (atanasovRazvan) [reply src/landing/LandingPage.js]: added "sources.js" file with the queries and mutations

File: src/landing/LandingPage.js (hunk 3 above)
Comment (atanasovRazvan) [reply src/landing/LandingPage.js]: updated

File: src/landing/LandingPage.js (hunk 2 above)
Comment (atanasovRazvan) [reply src/landing/LandingPage.js]: updated (modified logic)

**Review by ih90 — APPROVED** [review]

**Pull request merged**